ratatui-core/src/text/line.rs

```diff
@@ -179,18 +179,18 @@ use crate::widgets::Widget;
 ///
 /// [`Stylize`]: crate::style::Stylize
 #[derive(Default, Clone, Eq, PartialEq, Hash)]
-pub struct Line<'a> {
+pub struct Line<'lend, 'data> {
     /// The style of this line of text.
     pub style: Style,
 
     /// The alignment of this line of text.
     pub alignment: Option<Alignment>,
 
     /// The spans that make up this line of text.
-    pub spans: Vec<Span<'a>>,
+    pub spans: Cow<'lend, [Span<'data>]>,
```

Comment:
`Cow` is covariant in `T`, so I don't think we need two lifetimes in here - e.g. this works as-is:
```rust
use std::borrow::Cow;

struct Line<'a> {
    spans: Cow<'a, [Span<'a>]>,
}

#[derive(Clone)]
struct Span<'a> {
    str: Cow<'a, str>,
}

fn line<'a>(s: &'a str) -> Line<'a> {
    Line {
        spans: Cow::Owned(vec![Span { str: Cow::Borrowed(s) }]),
    }
}
```